src/Umbraco.Web.Website/Controllers/UmbLoginController.cs

```diff
@@ -33,12 +51,133 @@ public class UmbLoginController : SurfaceController
         IPublishedUrlProvider publishedUrlProvider,
         IMemberSignInManager signInManager,
         IMemberManager memberManager,
-        ITwoFactorLoginService twoFactorLoginService)
+        ITwoFactorLoginService twoFactorLoginService,
+        IEmailSender emailSender,
+        IHttpContextAccessor httpContextAccessor,
+        LinkGenerator linkGenerator,
+        IOptions<WebRoutingSettings> webRoutingSettings,
+        IOptions<GlobalSettings> globalSettings)
         : base(umbracoContextAccessor, databaseFactory, services, appCaches, profilingLogger, publishedUrlProvider)
     {
         _signInManager = signInManager;
         _memberManager = memberManager;
         _twoFactorLoginService = twoFactorLoginService;
+        _emailSender = emailSender;
+        _httpContextAccessor = httpContextAccessor;
+        _linkGenerator = linkGenerator;
+        _webRoutingSettings = webRoutingSettings.Value;
+        _globalSettings = globalSettings.Value;
+    }
+
+    [Obsolete("Use ctor with all params")]
+    public UmbLoginController(
+        IUmbracoContextAccessor umbracoContextAccessor,
+        IUmbracoDatabaseFactory databaseFactory,
+        ServiceContext services,
+        AppCaches appCaches,
+        IProfilingLogger profilingLogger,
+        IPublishedUrlProvider publishedUrlProvider,
+        IMemberSignInManager signInManager)
+        : this(
+            umbracoContextAccessor,
+            databaseFactory,
+            services,
+            appCaches,
+            profilingLogger,
+            publishedUrlProvider,
+            signInManager,
+            StaticServiceProvider.Instance.GetRequiredService<IMemberManager>(),
+            StaticServiceProvider.Instance.GetRequiredService<ITwoFactorLoginService>(),
+            StaticServiceProvider.Instance.GetRequiredService<IEmailSender>(),
+            StaticServiceProvider.Instance.GetRequiredService<IHttpContextAccessor>(),
+            StaticServiceProvider.Instance.GetRequiredService<LinkGenerator>(),
+            StaticServiceProvider.Instance.GetRequiredService<IOptions<WebRoutingSettings>>(),
+            StaticServiceProvider.Instance.GetRequiredService<IOptions<GlobalSettings>>())
+    {
+    }
+
+    [HttpPost]
+    [ValidateAntiForgeryToken]
+    [ValidateUmbracoFormRouteString]
+    public async Task<IActionResult> HandleForgottenPassword([Bind(Prefix = "forgottenPasswordModel")] ForgottenPasswordModel model)
+    {
+        if (ModelState.IsValid == false)
+        {
+            return CurrentUmbracoPage();
+        }
+
+        MergeRouteValuesToModel(model);
+
+        if (!_emailSender.CanSendRequiredEmail())
+        {
+            return CurrentUmbracoPage();
+        }
+
+        MemberIdentityUser? memberIdentity = await _memberManager.FindByNameAsync(model.Username);
+        if (memberIdentity == null!)
+        {
+            return new ValidationErrorResult(
+                $"No local member found for username {model.Username}");
```

Comment:
This is not a good security practice, you're giving a potential attacker information about which email addresses are / are not registered on this site. The best practice is to return a message like: "If this is a known account then you should receive instructions in your email shortly."

Reply:
I think I just copied this based on how it worked in backoffice users, when I think the `ValidationErrorResult` will show an error notification in backoffice.

Not sure if we should use this in frontend and return a different type.